internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -1551,3 +1616,100 @@ func (r *BMACReconciler) configureBMHForCleaning(ctx context.Context, log logrus
 	}
 	return
 }
+
+// nodeDrainTimedout returns true if the node drain should be skipped due to timout, false if draining should be run
+func nodeDrainTimedout(bmh *bmh_v1alpha1.BareMetalHost) (bool, error) {
+	annotations := bmh.GetAnnotations()
+	timeoutValue, timeoutExists := annotations[BMH_NODE_DRAIN_TIMEOUT_ANNOTATION]
+	startValue, startExists := annotations[BMH_NODE_DRAIN_START_ANNOTATION]
+
+	if !startExists || !timeoutExists {
+		return false, nil
+	}
+
+	drainTimeout, err := time.ParseDuration(timeoutValue)
+	if err != nil {
+		return false, err
+	}
+	firstTimeDrain, err := time.Parse(time.RFC3339, startValue)
+	if err != nil {
+		return false, err
+	}
+	now := time.Now().UTC()
+	diff := now.Sub(firstTimeDrain)
+	return diff.Seconds() >= drainTimeout.Seconds(), nil
+}
+
+func nodeUnreachable(node *corev1.Node) bool {
+	if node == nil {
+		return false
+	}
+	for _, c := range node.Status.Conditions {
+		if c.Type == corev1.NodeReady {
+			return c.Status == corev1.ConditionUnknown
+		}
+	}
+	return false
+}
+
+// Mostly taken from https://github.com/kubernetes-sigs/cluster-api/blob/539760a/internal/controllers/machine/machine_controller.go#L586
+func (r *BMACReconciler) drainAgentNode(ctx context.Context, log logrus.FieldLogger, agent *aiv1beta1.Agent) (bool, error) {
+	log = log.WithFields(logrus.Fields{
+		"spoke_cluster_name":      agent.Spec.ClusterDeploymentName.Name,
+		"spoke_cluster_namespace": agent.Spec.ClusterDeploymentName.Namespace,
+	})
+	// get spoke connection
```

Comment:
Add log saying "Draining agent node"

Reply:
I have a log in the caller, but I can move it here if you prefer.